ci/common-ci.yml

```
@@ -34,7 +34,7 @@ stages:
     reports:
       dotenv: build.env
   variables:
-    SPACK_SHA: dcc4423a9d0219a26ceeb52e329ed956e41f4d2c
+    SPACK_SHA: fd3c18b6fd7eda8abbb72ee81b39aa81b053d9bd
```

```suggestion
    SPACK_SHA: 1184de8352281ce1b34a8f3f6b54d7fadc2b216a
```
Worth testing with https://github.com/spack/spack/pull/44242?

---

I think it's worth testing, but I'll do that in a follow-up PR.

---

I would postpone this check after the release in a new PR.
If the new one-api package doesn't work, it need to be fixed in spack.

---

Since there were still other changes to be made, I also bumped the spack commit on this PR.

---

The new commit also works well. On the old commit the gfortran mkl was linked, on the new commit the intel one, so all good there!